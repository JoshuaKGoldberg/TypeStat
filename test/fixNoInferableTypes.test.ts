import path from "node:path";
import { describe, expect, it } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

describe("No inferable types", () => {
	it("parameters", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/noInferableTypes/parameters",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("property declarations", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/noInferableTypes/propertyDeclarations",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("variable declarations", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/noInferableTypes/variableDeclarations",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});
});
