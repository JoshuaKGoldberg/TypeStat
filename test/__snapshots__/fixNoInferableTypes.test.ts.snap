// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`No inferable types > parameters > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "files": [
      "actual.ts"
    ],
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": true
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": false,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": true,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "mutators": [
    [
      "fixImportExtensions",
      null
    ],
    [
      "fixIncompleteTypes",
      null
    ],
    [
      "fixMissingProperties",
      null
    ],
    [
      "fixNoImplicitAny",
      null
    ],
    [
      "fixNoImplicitThis",
      null
    ],
    [
      "fixNoInferableTypes",
      null
    ],
    [
      "fixStrictNonNullAssertions",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": []
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {
    "strictNullChecks": true
  }
}"
`;

exports[`No inferable types > property declarations > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "files": [
      "actual.ts"
    ],
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": true
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": false,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": true,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "mutators": [
    [
      "fixImportExtensions",
      null
    ],
    [
      "fixIncompleteTypes",
      null
    ],
    [
      "fixMissingProperties",
      null
    ],
    [
      "fixNoImplicitAny",
      null
    ],
    [
      "fixNoImplicitThis",
      null
    ],
    [
      "fixNoInferableTypes",
      null
    ],
    [
      "fixStrictNonNullAssertions",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": []
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {
    "strictNullChecks": true
  }
}"
`;

exports[`No inferable types > variable declarations > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "files": [
      "actual.ts"
    ],
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": true
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": false,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": true,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "mutators": [
    [
      "fixImportExtensions",
      null
    ],
    [
      "fixIncompleteTypes",
      null
    ],
    [
      "fixMissingProperties",
      null
    ],
    [
      "fixNoImplicitAny",
      null
    ],
    [
      "fixNoImplicitThis",
      null
    ],
    [
      "fixNoInferableTypes",
      null
    ],
    [
      "fixStrictNonNullAssertions",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": []
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {
    "strictNullChecks": true
  }
}"
`;
